### PR TITLE
UW-455

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Description**

Expand on _Documentation_ section in _Contributor Guide_. Preview docs [here](https://unified-workflow--369.org.readthedocs.build/en/369/).

**Type**

- [x] Documentation

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: docs/sections/contributor_guide/developer_setup.rst [hunk 1]
```diff
@@ -25,8 +25,9 @@ If the above is successful, you will be in a ``workflow-tools`` development shel
 
 Future ``make devshell`` invocations will be almost instantaneous, as the underlying virtual environment will already exist. In general, all source code changes will be immediately live in the development shell, subject to execution, test, etc. But some changes – especially to the contents of the ``recipe/`` directory, or to the ``src/setup.py`` module – may require recreation of the development shell. If you know this is needed, or when in doubt:
 
-  1. Exit the development shell, run ``conda env remove -n DEV-uwtools`` to remove the old environment.
-  2. Run ``make devshell`` to recreate it.
+  #. Exit the development shell.
+  #. Run ``make clean-devenv`` (or ``conda env remove -n DEV-uwtools``) to remove the old environment.
+  #. Run ``make devshell`` to recreate it.
```
Comment (maddenp-noaa) [inline docs/sections/contributor_guide/developer_setup.rst:30]: Let Sphinx number the bullet points, and add advice about the new `make clean-devenv` target.

File: docs/sections/contributor_guide/index.rst [hunk 2]
```diff
@@ -2,6 +2,8 @@ Contributor Guide
 =================
 
 .. toctree::
+   :maxdepth: 2
```
Comment (maddenp-noaa) [inline docs/sections/contributor_guide/index.rst:5]: This makes the tables of contents a bit more compact and IMO easier to read/use.

File: docs/sections/contributor_guide/developer_setup.rst
Comment (elcarpenterNOAA) [inline docs/sections/contributor_guide/developer_setup.rst:1]: On line 43, "available" is spelled incorrectly.

**Review by christinaholtNOAA — APPROVED** [review]
This is great information!

File: docs/sections/contributor_guide/developer_setup.rst [hunk 3]
```diff
@@ -25,8 +25,9 @@ If the above is successful, you will be in a ``workflow-tools`` development shel
 
 Future ``make devshell`` invocations will be almost instantaneous, as the underlying virtual environment will already exist. In general, all source code changes will be immediately live in the development shell, subject to execution, test, etc. But some changes – especially to the contents of the ``recipe/`` directory, or to the ``src/setup.py`` module – may require recreation of the development shell. If you know this is needed, or when in doubt:
 
-  1. Exit the development shell, run ``conda env remove -n DEV-uwtools`` to remove the old environment.
-  2. Run ``make devshell`` to recreate it.
+  #. Exit the development shell.
+  #. Run ``make clean-devenv`` (or ``conda env remove -n DEV-uwtools``) to remove the old environment.
```
Comment (christinaholtNOAA) [inline docs/sections/contributor_guide/developer_setup.rst:29]: I appreciate you helping me out here!

**Review by elcarpenterNOAA — APPROVED** [review]
LGTM.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
LGTM!

**Pull request merged**